src/main/java/org/example/auth/JwtAuthenticator.java

```diff
@@ -0,0 +1,34 @@
+package org.example.auth;
+
+import io.dropwizard.auth.Authenticator;
+import io.jsonwebtoken.Jwts;
+import org.example.models.JwtToken;
+import org.example.models.UserRole;
+
+import java.security.Key;
+import java.util.Optional;
+
+public class JwtAuthenticator implements Authenticator<String, JwtToken> {
+    Key key;
+
+    public JwtAuthenticator(final Key key) {
+        this.key = key;
+    }
+
+    @Override
+    public Optional<JwtToken> authenticate(final String token) {
+        try {
+            Integer roleId = Jwts.parser()
+                    .setSigningKey(key)
+                    .build()
+                    .parseSignedClaims(token)
```

Comment:
double check the latest implementation for this, parseJWSClaim is an alternative but may not be in this version. Just have a wee look
https://stackoverflow.com/questions/76973793/what-to-put-instead-of-parseclaimsjws-for-my-jwtgenerator-for-my-spring-securi